distribution/lib/Standard/Base/0.0.0-dev/src/Exception.enso

```diff
@@ -0,0 +1,95 @@
+import project.Any.Any
+import project.Data.Text.Text
+import project.Error.Error
+import project.Meta
+import project.Panic.Panic
+from project.Data.Boolean import Boolean, False, True
+from project.Runtime import assert
+
+## PRIVATE
+   `Exception` is not a type of error; rather, it is a way to throw an `Error`
+   in imperative-style code and ensure that it is always handled or propagated
+   to the top level.
+
+   `throw_exception` works by wrapping the error in an `Uncaught_Exception` and
+   throwing it with `Panic.throw`. This should occur inside a `catch_exception`
+   block, which catches the `Panic` and rethrows the wrapped error as a dataflow
+   error.
+
+    > Example
+      Abort an imperative block with `throw_exception`.
+
+          handler e =
+              IO.println 'Caught: '+e.to_text
+
+          Exception.catch_exception My_Error handler=handler <|
+              a = 10
+              b = a + a
+              IO.println a
+              IO.println b
+              Exception.throw_exception (My_Error.Error "error")
+              IO.println "Does not reach here"
+
+          # Prints:
+          # 10
+          # 20
+          # Caught: (My_Error.Error 'error')
+type Exception
```

Comment:
Is this not a very similar concept to `Panic.handle_wrapped_dataflow_error` and `Panic.throw_wrapped_if_error`?

Could we merge the two?

Reply:
Yes, good idea.